Feat: Responsive tooltips

miksa1987 opened the pull request:
All right, here's a fixed PR discussed here: [#157](https://github.com/neherlab/covid19_scenarios/pull/157)

There was some weirdness with rebasing. So this PR should only deal with my changes, which include:

- Custom ResponsiveTooltipContent component, which will divide in two columns on mobile for better readability.
- Repositioning of the chart Tooltip component when in single column mode.
- When clicking on chart in single column mode, the page scrolls the chart to top, so that repositioned Tooltip is wholly visible.

@ivan-aksamentov if you're happy with this you can merge.

rcbevans commented:
Tooltip values do not appear to be internationalized/humanized:

![image](https://user-images.githubusercontent.com/2041260/77706096-a7343380-6f7e-11ea-94e8-8fd749834d90.png)

rcbevans commented on src/components/Main/Results/ResponsiveTooltipContent.tsx:
```diff
@@ -0,0 +1,55 @@
+            {left.map((item, idx) => <ToolTipContentItem key={idx} name={item.name} value={item.value} color={item.color} />)}
+          </div>
+          <div>
+            {right.map((item, idx) => <ToolTipContentItem key={idx} name={item.name} value={item.value} color={item.color} />)}
```
I believe this is the `item.value` that needs to be appropriately formatted for display.

rcbevans commented on src/components/Main/Results/DeterministicLinePlot.tsx:
```diff
@@ -230,7 +239,11 @@ export function DeterministicLinePlot({ data, userResult, logScale, showHumanize
                 <YAxis scale={logScaleString} type="number" domain={[1, 'dataMax']} tickFormatter={yTickFormatter} />
-                <Tooltip formatter={tooltipFormatter} labelFormatter={labelFormatter} />
+                <Tooltip 
+                  formatter={tooltipFormatter} 
```
You need to pass this tooltipFormatter to your ResponsiveTooltipContent and format the right hand value as necessary.  I don't think the tooltip is using the formatter, since custom content takes priority.

miksa1987 commented:
@ivan-aksamentov number formatting should work now as intended. The ResponsiveTooltipContent now formats the values with the formatter if passed to it. 

However, I'm wondering about all the parameters tooltipFormatter needs, if it only returns the formatted value. Should it be simplified to take only the value and format that, or is there a thought on expanding it's capabilities that I should take into account on my component?

gj262 commented on src/components/Main/Results/DeterministicLinePlot.tsx:
```diff
@@ -230,7 +239,11 @@ export function DeterministicLinePlot({ data, userResult, logScale, showHumanize
-                <Tooltip formatter={tooltipFormatter} labelFormatter={labelFormatter} />
+                <Tooltip 
+                  formatter={tooltipFormatter} 
+                  position={tooltipPosition} 
```
Can you pass `labelFormatter={labelFormatter}` also so that `<ResponsiveTooltipContent />` does not have to guess as to label type i.e. date -v- age.

gj262 commented on src/components/Main/Results/ResponsiveTooltipContent.tsx:
```diff
@@ -0,0 +1,58 @@
+
+export function ResponsiveTooltipContent({ active, payload, label, formatter }: TooltipContentProps) {
+  const formattedLabel = Number(label) > 10000 ? new Date(label).toISOString().slice(0, 10) : label
+  const formatNumber = formatter
```
could use labelFormatter here (I think)

miksa1987 commented:
> Looks great. See comment re: label formatting.

Thanks for noting that. I updated my component to use the labelFormatter and passed it in DeterministicLinePlot. I also rebased the whole thing.

ivan-aksamentov commented:
Excellent job!
